index.bs

```diff
@@ -424,58 +433,150 @@ When this method is invoked, the user agent MUST execute the following algorithm
+    cancel the timer started in step 2, return [=a promise rejected with=] with
+    a {{DOMException}} whose name is "{{NotSupportedError}}", and terminate this
+    algorithm.
+
+1. Let |clientExtensions| be a new [=list=].
+1. If the {{ScopedCredentialOptions/extensions}} member of {{options}} is
+    <a>present</a>, then [=map/for each=] |extension| → |argument| of
+    <code>{{options}}.{{ScopedCredentialOptions/extensions}}</code>:
+    1. If |extension| is not supported by this client platform, then either:
+        * [=Continue=], or
+        * Let |result| be a CBOR ([[!RFC7049]]) encoding of |extension|.
+
+            Issue(w3c/webauthn#363): Define this encoding more precisely.
+    1. Otherwise, let |result| be the result of running |extension|'s [=client
+        processing=] algorithm on |argument|. If the algorithm returned an
+        error, [=continue=].
+
+        Issue(w3c/webauthn#363): Ensure all extensions define a client
+        processing algorithm.
+    1. [=list/Append=] |result| to |clientExtensions|.
+
+1. Let |clientData| be a new {{ClientData}} instance whose fields are:
+    : {{challenge}}
+    :: The [=base64url encoding=] of {{attestationChallenge}}
+    : {{origin}}
+    :: The [=unicode serialization of an origin|unicode serialization=] of |rpId|
+    : {{hashAlg}}
+    :: UA-chosen
+
+        Issue(w3c/webauthn#362): We need *some* constraints on the possible hash
+        algorithms, or else sites will fail on unusual UAs.
+    : {{tokenBinding}}
+    :: The [=Token Binding ID=] associated with |callerOrigin| (if any)
+
+        Issue(w3c/webauthn#360): Make sure this association was set up properly.
+    : {{ClientData/extensions}}
+    :: |clientExtensions|
+
+1. Let |clientDataJSON| be the [=UTF-8 encoding=] of the result of calling the
+    initial value of {{JSON/stringify|JSON.stringify}} on |clientData|.
+
+    Issue: Some extensions contain ArrayBuffers, which don't stringify well.
+    What's the intent here?
+1. Let |clientDataHash| be the hash of |clientDataJSON| using
+    <code>|clientData|.{{hashAlg}}</code>.
+
+1. Let |issuedRequests| and |currentlyAvailableAuthenticators| be new [=ordered
+    sets=].
+
+1. For each |authenticator| currently available on this platform, if
+    <code>{{options}}.{{ScopedCredentialOptions/attachment}}</code> is not
+    [=present=] or its value matches |authenticator|'s attachment modality,
+    [=set/append=] |authenticator| to |currentlyAvailableAuthenticators|.
+
+1. [=set/For each=] |authenticator| in |currentlyAvailableAuthenticators|:
+    1. Let |excludeList| be a new [=list=].
+    1. [=list/For each=] credential |C| in <code>{{options}}.{{ScopedCredentialOptions/excludeList}}</code>:
+        1. If |C| has an empty {{transports}} list, [=list/append=] |C| to
+            |excludeList| and [=continue=].
+        1. If |authenticator| is connected over a transport not mentioned in
```

Comment:
This is an "else" from the previous point, right? This seems a bit unwieldy.

Perhaps we could replace all these three sub-bullets with two:
1. If |C|.{{transports}} is non-empty and does not contain the transport that |authenticator| is connected over, the client MAY [=continue=].
1.  [=list/Append=] |C| to |excludeList|.